tests/unit/proxy/abstract_proxy_tests.py

```diff
@@ -42,6 +61,216 @@ def tearDown(self) -> None:
     def get_proxy(self) -> T:
         pass
 
+    @abstractmethod
+    def get_relationship(self, *, node_type1: str, node_key1: str, node_type2: str,
+                         node_key2: str) -> List[Any]:
+        pass
+
+    def test_rt_table(self) -> None:
+        """
+        it'd be nice to check that the result could be deserialized as a client of the metadata_service would
+        """
+        # TODO: check fixtures
+        expected = Fixtures.next_table()
+        expected.description = '"hello!" said no one'
+        expected.tags.sort()
+
+        self.get_proxy().put_table(table=expected)
+        actual: Table = self.get_proxy().get_table(table_uri=checkNotNone(expected.key))
+        actual.last_updated_timestamp = None
+        actual.tags.sort()
+
+        self.assertEqual(expected, actual)
+
+    def test_rt_table_with_owner_user_not_application(self) -> None:
+        user = Fixtures.next_user(is_active=True)
+        self.get_proxy().put_user(data=user)
+        application = Fixtures.next_application(application_id=user.user_id)
+        expected = Fixtures.next_table(application=application)
+        self.get_proxy().put_table(table=expected)
+
+        actual: Table = self.get_proxy().get_table(table_uri=checkNotNone(expected.key))
+
+        self.assertEqual(user.user_id, actual.owners[0].user_id)
+
+    def test_rt_table_with_non_existent_app(self) -> None:
+        application = Fixtures.next_application()
+        # purposefully don't insert application
+        expected_table = Fixtures.next_table(application=application)
+
+        self.get_proxy().put_table(table=expected_table)
+        actual_table: Table = self.get_proxy().get_table(table_uri=checkNotNone(expected_table.key))
+
+        self.assertEqual(actual_table.table_writer, None)
+        self.assertEqual(actual_table.owners, [])
+
+    def test_expire_columns(self) -> None:
```

Comment:
I suppose we could just support expiry in gremlin (even if its not in other proxies)

Reply:
Probably best to introduce as a feature request and add this back in then, I think.

Thanks for the review, @alran!